app/containers/Products/LinkedProductModal.tsx

```diff
@@ -0,0 +1,179 @@
+  setLinkProductModalShow: (boolean) => void;
+  saveLinkedProducts: (...args: [any]) => void;
+  query: any;
+  linkData: any;
+}
+
+export const LinkedProductModalComponent: React.FunctionComponent<Props> = ({
+  product,
+  linkProductModalShow,
+  setLinkProductModalShow,
+  saveLinkedProducts,
+  query,
+  linkData
+}) => {
+  const CUSTOM_FIELDS = {
+    productRowId: (props) => {
+      return (
+        <ProductRowIdField
+          isLinkModal
+          query={props.formContext.query}
+          {...props}
+        />
+      );
+    }
+  };
+
+  const linkSchema: JSONSchema6 = {
+    type: 'array',
+    items: {
+      $ref: '#/definitions/product'
+    },
+    definitions: {
+      product: {
+        type: 'object',
+        properties: {
+          productRowId: {
+            type: 'integer'
+          }
+        }
+      }
+    }
+  };
+
+  const linkUISchema = {
+    'ui:add-text': '+',
+    'ui:remove-text': '-',
+    linkProduct: true,
+    items: {
+      'ui:field': 'product',
+      classNames: 'hidden-title',
+      productRowId: {
+        'ui:col-md': 12,
+        'ui:widget': 'SearchWidget',
+        'ui:field': 'productRowId'
+      }
+    }
+  };
+
+  return (
+    <Modal
+      centered
+      size="xl"
+      show={linkProductModalShow}
+      onHide={() => {
+        setLinkProductModalShow(false);
+      }}
+    >
+      <Modal.Header closeButton style={{color: 'white', background: '#003366'}}>
+        <Modal.Title>Product Associations</Modal.Title>
+      </Modal.Header>
+      <Modal.Body style={{background: '#f5f5f5'}}>
+        <Container>
+          <Row>
+            <Col md={12}>
+              <h5>Add/Edit Associations</h5>
+              <hr />
+              <p>
+                When you specify one or more products on the right, applications
+                reporting the product on the left will be prompted to report
+                these products as well.
+              </p>
+              <p>
+                Press the &apos;+&apos; button and select a product to add an
+                association, press the &apos;-&apos; button to remove an
+                association. Deleting an association will not delete the
+                products themselves.
+              </p>
+              <p>
+                Once you have created all the necessary associations, press
+                &apos;Save&apos; to apply them.
+              </p>
+              <hr />
+              <br />
+            </Col>
+          </Row>
+          <Row>
+            <Col md={3}>
+              <h4>Product Name:</h4>
+            </Col>
+            <Col md={9}>
+              <h4>Associated Products:</h4>
+            </Col>
+          </Row>
+          &emsp;
+          <Row>
+            <Col md={3}>
+              <h5>{product.productName}</h5>
+            </Col>
+            <Col md={9}>
+              <JsonSchemaForm
+                showErrorList={false}
+                ArrayFieldTemplate={FormArrayFieldTemplate}
```

Comment:
You can make a custom ArrayFieldTemplate if you need instead of changing the generic one

Reply:
I did create a custom LinkedProductFormArrayFieldTemplate at first, but it was so similar to the original that the sonarcloud code duplication check was angry, which is why I added the logic to the FormArrayFieldTemplate

Reply:
Maybe let's just get this merged as-is and open a TechDebt card to refactor this.

Reply:
https://youtrack.button.is/issue/GGIRCS-1648